stress_test/controller.py

```
@@ -0,0 +1,338 @@
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :rtype int
+        :type timeout_ms: int
+        """
+
+        logging.info('[Controller] Waiting to start listening on a port')
+
+        timeout = time.time() + (float(timeout_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            gpid = util.process.getpid_listeningonport(self.port, self.ssh_pass)
+            logging.info('Returned pid listening on port {0}: {1}'.
+                          format(self.port, gpid))
+            if gpid == 0:
+                raise Exception('Another controller seems to have started in the '
+                                'meantime. Exiting...')
+
+        raise Exception('Controller failed to start within a period of {0} '
+                        'minutes'.format(timeout))
+
+
+    def wait_until_up(self,timeout_ms):
+        """ Waits for controller status to become 1 (started).
+
+        :param timeout_ms: milliseconds to wait (in milliseconds).
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :type timeout_ms: int
+        """
+
+        logging.info('[Controller] Waiting to be started')
+
+        timeout = time.time() + (float(timeout_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            if check_status() == '1':
+                return
+
+        raise Exception('Controller failed to start. '
+                        'Status check returned 0 after trying for {0} seconds.'.
+                        format(float(timeout_ms) / 1000))
+
+
+    @staticmethod
+    def new_controller(self)
+        """ Factory method. Creates a subclass class depending on the controller name
+        :returns: a subclass or None
+        :rtype: failed_flow_ops int
+        """
+
+        if (self.name == 'OLD'):
```

1. typo: ~~OLD~~ ODL
2. pass constructor args here, i.e. `ODL(ctrl_base_dir, test_config)`
3. better to raise an Exception both for ONOS and for an unknown controller name, to let the user know that we do not support them yet